tests/test_time_machine.py

```diff
@@ -434,6 +434,36 @@ def test_destination_datetime_tzinfo_zoneinfo_no_orig_tz():
         with time_machine.travel(dest):
             assert time.tzname == ("EAT", "EAT")
 
+        dest2 = LIBRARY_EPOCH_DATETIME.replace(tzinfo=ZoneInfo("UTC"))
+
+        with time_machine.travel(dest2):
+            assert time.tzname == ("UTC", "UTC")
+
+        assert time.tzname == orig_tzname
+
+
+def test_destination_datetime_tzinfo_datetime_timezone_utc_no_orig_tz():
+    with change_local_timezone(None):
+        orig_tzname = time.tzname
+        dest = LIBRARY_EPOCH_DATETIME.replace(tzinfo=dt.timezone.utc)
+
+        with time_machine.travel(dest):
+            assert time.tzname == ("UTC", "UTC")
+
+        assert time.tzname == orig_tzname
```

Comment:
Not 100% sure if necessary to repeat this last assertion with `test_destination_datetime_tzinfo` tests. Probably good to keep in just for peace of mind the original timezone is restored.